File: 443_StringCompression.cpp
class Solution {
public:
    int compress(vector<char>& chars) {
        
        
        vector<char> dummy;
        int count = 1;
        char current;
        char prev;
        dummy.push_back(chars.at(0));
            
        for(int i = 1; i < chars.size(); i++)
        {
            current = chars.at(i);
            prev = chars.at(i-1);
            
            if(current == prev)
            {
                count++;
            }
            else
            {
                if(count > 1)
                {
                    if(count > 9)
                    {
                        int number = count;
                        vector<int> numberList;
                        
                        while(number)
                        {
                            numberList.push_back(number % 10);
                            number = number / 10;
                        }
                        
                        //numberList.reverse();
                        reverse(numberList.begin(), numberList.end());
                        
                        for(int j = 0; j < numberList.size(); j++)
                        {
                            dummy.push_back('0' + numberList.at(j));
                        }
                        count = 1;
                        
                    }
                    else
                    {
                        dummy.push_back('0' + count);
                        count = 1;
                    }
                }
                dummy.push_back(current);
            }
        }
        
        if(count > 1)
        {
            if(count > 9)
            {
                        int number = count;
                        vector<int> numberList;
                        
                        while(number)
                        {
                            numberList.push_back(number % 10);
                            number = number / 10;
                        }
                        
                        //numberList.reverse();
                        reverse(numberList.begin(), numberList.end());
                        
                        for(int j = 0; j < numberList.size(); j++)
                        {
                            dummy.push_back('0' + numberList.at(j));
                        }
                
            }
            else
            {
                dummy.push_back('0' + count);
                count = 1;
            }
        }
        
        for(int i = 0; i < dummy.size(); i++)
        { 
            chars.at(i) = dummy.at(i);
        }
        
        return dummy.size();
        
        
    }
};

File: 443_StringCompression.py
class Solution:
    def compress(self, chars: List[str]) -> int:
        
        count = 1
        output = []
        
        output.append(chars[0])
        
        for i in range(1,len(chars)):
        
            current = chars[i]
            prev = chars[i-1]
            
            if current == prev:
                count += 1
            else:
                if count > 1:
                    if count > 9:
                        numberCount = str(count)
                        for i in range(len(numberCount)):
                            output.append(numberCount[i])
                        # tensplace = count // 10
                        # unitsplace = count % 10
                        # output.append(str(tensplace))
                        # output.append(str(unitsplace))
                    else:
                        output.append(str(count))
                    count = 1
                
                output.append(current)
            
        
        if count > 1:
            if count > 9:
                numberCount = str(count)
                for i in range(len(numberCount)):
                    output.append(numberCount[i])
                # tensplace = count // 10
                # unitsplace = count % 10
                # output.append(str(tensplace))
                # output.append(str(unitsplace))
            else:
                output.append(str(count))
        
        print(len(output))
        print(output)
        
        for i in range(len(output)):
            chars[i] = output[i]
        
        return len(output)
            
    